src/main/java/org/magicdgs/readtools/utils/mappability/gem/GemMappabilityReader.java

```diff
@@ -0,0 +1,304 @@
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.NoSuchElementException;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Iterator over gem-mappability number of mappings.
+ *
+ * <p>The GEM-mappability FASTA-like format consists on:
+ * <ul>
+ *     <li>Header with metadata. Includes information about the parameters used and the encoding for
+ *     the values of mappability. See {@link GemMappabilityHeader} for more information</li>
+ *     <li>Per-base range of values. Encoded as a char, representing a range of number of mappings.</li>
+ * </ul>
+ *
+ * <p>This class iterates over the file and retrieves the per-base range of values as a
+ * {@link GemMappabilityRecord}, which contains the sequence name, position and range of values.
+ *
+ * @author Daniel Gomez-Sanchez (magicDGS)
+ */
+public final class GemMappabilityReader implements CloseableIterator<GemMappabilityRecord> {
+
+    // HEADER PREFIX (for sequence name and meta-data)
+    private static final char HEADER_PREFIX = '~';
+
+    ///////////////////////////
+    // HEADER LINES - prefixed by two HEADER_PREFIX
+
+    private static final String KMER_LENGTH_HEADER_TEXT = "K-MER LENGTH";
+    private static final String APPROX_THRESHOLD_HEADER_TEXT = "APPROXIMATION THRESHOLD";
+    private static final String MAX_MISMATCH_HEADER_TEXT = "MAX MISMATCHES";
+    private static final String MAX_ERRORS_HEADER_TEXT = "MAX ERRORS";
+    private static final String MAX_INDEL_LENGTH_HEADER_TEXT = "MAX BIG INDEL LENGTH";
+    private static final String MIN_MATCH_HEADER_TEXT = "MIN MATCHED BASES";
+    private static final String STRATA_AFTER_BEST_HEADER_TEXT = "STRATA AFTER BEST";
+    private static final String ENCODING_HEADER_TEXT = "ENCODING";
+
+    // pattern for the encoding (char to value range)
+    protected static final Pattern ENCODING_PATTERN = Pattern.compile("'(.)'~\\[(\\d+)-(\\d+)]");
+
+    // stored for error message and get the identity of the path
+    private final Path path;
+    // reader (BLR allows to peek and track the line number for error messages)
+    private final BufferedLineReader reader;
+    // header from the file, read on construction
+    private final GemMappabilityHeader header;
+
+    // iteration values to keep track of the current sequence
+    private String currentSequence = null;
+    private int currentSequencePosition = -1;
+
+    // logger
+    private final Logger logger = LogManager.getLogger(this.getClass());
+
+    /**
+     * Loads a gem-mappability formatted file.
+     *
+     * @param path path where the file is located.
+     * @throws IOException if the reader cannot be open.
+     */
+    public GemMappabilityReader(final Path path) throws IOException {
+        this.path = Utils.nonNull(path);
+        this.reader = new BufferedLineReader(Files.newInputStream(this.path));
```

Comment:
Even if there is no output in GEM-mappability from a gzip/bgzip output, compressing might be useful to store the data (as it is redundant) and it will minimize the file size while still useful for this tool. Add an issue for that!